internal/controller/components/modelregistry/modelregistry_controller.go

```diff
@@ -92,12 +86,10 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 		)).
 		WithAction(deployments.NewAction()).
 		WithAction(updateStatus).
-		// must be the final action
-		WithAction(gc.NewAction(
-			gc.WithUnremovables(gvk.ServiceMeshMember),
-		)).
 		// declares the list of additional, controller specific conditions that are
 		// contributing to the controller readiness status
+		// must be the final action
```

Comment:
Nit: moving the `WithAction(gc.NewAction())` here, along with its comment, made these 3 lines of comments a little bit incoherent: the first two lines are for the `WithConditions(conditionTypes...)` below the gc part, and then the 3rd comment line is the one for the gc line.